src/main/java/com/celements/course/service/CourseService.java

```diff
@@ -428,7 +452,10 @@ private boolean validateParticipant(String activationCode, BaseObject partiObj)
     String hashedCode = passwordHashString(activationCode);
     String savedHash = partiObj.getStringValue("validkey");
     if (hashedCode.equals(savedHash)) {
-      partiObj.setStringValue("status", "confirmed");
+      ParticipantStatus state = ParticipantStatus.valueOf(partiObj.getStringValue("status"));
+      if (!DEFAULT_IGNORE_STATES.contains(state)) {
+        partiObj.setStringValue("status", "confirmed");
```

Comment:
please use XObjectFieldAccessor (as requirement) to get/set fields from an object:
`fieldAccessor.getValue(obj, field)`
`fieldAccessor.setValue(obj, field, value)`